src/Coordination/KeeperStorage.cpp

```diff
@@ -865,6 +915,16 @@ struct KeeperStorageGetRequestProcessor final : public KeeperStorageRequestProce
             }
         }
 
+        // We cannot store the node because the result should be connected to the binary itself
+        // this way we avoid incorrect results when we read a snapshot from older Keeper that can have
+        // lower API version
```

Comment:
Maybe we should not write system node and its children to snapshot at all and maintain them only in memory?

Reply:
Well the problem is the digest and `numChildren == children.size()` assert, but I forgot that in the future we could delete a node and if we pull old system nodes from snapshot it would be just trash.
I'll push the code where we store only the system node path to preserve the digest and assert (we assume this won't be touched), but not the children because they can be changed and are binary dependent.